index.js

```diff
@@ -18,34 +18,39 @@ module.exports = function (name, defaults, argv, parse) {
     ? cc.json(defaults) : defaults
     ) || {}
 
-  parse = parse || cc.parse
+  parsers = parsers || cc.parsers
 
   var env = cc.env(name + '_')
-
   var configs = [defaults]
   var configFiles = []
   function addConfigFile (file) {
     if (configFiles.indexOf(file) >= 0) return
     var fileConfig = cc.file(file)
     if (fileConfig) {
-      configs.push(parse(fileConfig))
+      configs.push(cc.parse(fileConfig, file, parsers))
```

Comment:
I kind of think that the parse function should be created inside the main function. I did it this way so that `test/ini.js` still passed.